torch_xla/experimental/xla_quantized_matmul.py

```diff
@@ -14,17 +14,59 @@ def _check_per_channel_quant_weight_dtype_shapes(input_dim, output_dim, w,
   assert w.dtype == torch.int8, f"Weight dtype is expected to be torch.int8, got {w.dtype}."
   assert w.dim(
   ) == 2, f"Weight tensor is expected to be 2D, got {w.dim()}D Tensor."
-  assert output_dim == w.shape[0] and input_dim == w.shape[
-      1], f"Weight shape is expected to be [output_dim, input_dim], output_dim: {output_dim}, input_dim: {input_dim}, but got {w.shape}."
-  assert w_scaler.dim() == 1 and w_scaler.shape[0] == w.shape[
-      0], f"weight scaler shape is expect to be [out_channel,], got {w_scaler.shape}, weight shape {w.shape}."
+  w_shape = list(w.shape)
+  assert output_dim == w_shape[0] and input_dim == w_shape[
+      1], f"Weight shape is expected to be [output_dim, input_dim], output_dim: {output_dim}, input_dim: {input_dim}, but got {w_shape}."
+  assert w_scaler.dim() == 1 and w_scaler.shape[0] == w_shape[
+      0], f"weight scaler shape is expect to be [out_channel,], got {w_scaler.shape}, weight shape {w_shape}."
+
+
+def pack_4bit(x, dtype=torch.int8):
+  # Not used now, we don't do packed weight because reinterpret_cast
+  # doesn't work on TPU.
+  # Pack 4 bit tensor into (u)int8 or (u)int32
+  # Always pack along the last dim
+  assert dtype == torch.int8, "Only int8 pack/unpack is supported now."
+  num_int4_per_element = 2 if (dtype == torch.int8 or
+                               dtype == torch.uint8) else 8
```

Comment:
it will never be 8 right? you just assert on dtype above..

Reply:
Let me remove the pack unpack logic since not used now.